src/overlays/actors/ovl_Obj_Syokudai/z_obj_syokudai.c

```diff
@@ -27,11 +29,243 @@ const ActorInit Obj_Syokudai_InitVars = {
+                if (itemType < 0) {
+                    if (player->unk_860 == 0) {
+                        player->unk_860 = 210;
+                        Audio_PlaySoundGeneral(NA_SE_EV_FLAME_IGNITION, &this->actor.projectedPos, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+                    } else if (player->unk_860 < 200) {
+                        player->unk_860 = 200;
+                    }
+                } else if (toucherFlags & 0x20) {
+                    arrow = (EnArrow*) this->collider2.base.ac;
+                    if(arrow->actor.update && (arrow->actor.id == 0x16)) {
+                        arrow->actor.params = 0;
+                        arrow->unk_1A8 = 0x800;
+                    }
+                }
+                if ((0 <= this->timer) && (this->timer < (50 * litTimeScale + 100)) && (torchType != 0)) {
+                    this->timer = 50 * litTimeScale + 100;
+                }
+            } else if ((torchType != 0) && (((itemType > 0) && (toucherFlags & 0x20800))
+                                            || ((itemType < 0) && (player->unk_860 != 0)))) {
+                
+                if ((itemType < 0) && ( player->unk_860 < 200)){
+                    player->unk_860 = 200;
+                }
+                if (torchCount == 0) {
+                    this->timer = -1;
+                    if (torchType != 2) {
+                        Flags_SetSwitch(globalCtx, switchFlag);
+                        func_80080480(globalCtx, &this->actor);
+                    }
+                } else {
+                    sLitTorchCount++;
+                    if (sLitTorchCount >= torchCount) {
+                        Flags_SetSwitch(globalCtx, switchFlag);
+                        func_80080480(globalCtx, &this->actor);
+                        this->timer = -1;
+                    } else {
+                        this->timer = (litTimeScale * 50) + 110;
+                    }
+                }
+                Audio_PlaySoundGeneral(NA_SE_EV_FLAME_IGNITION, &this->actor.projectedPos, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+            }
+        }
+    }
+    collider1 = &this->collider1;
+    Collider_CylinderUpdate(&this->actor, collider1);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &collider1->base);
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &collider1->base);
+    
+    collider2 = &this->collider2;
+    Collider_CylinderUpdate(&this->actor, collider2);
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &collider2->base);
+
+    if (this->timer > 0) {
+        this->timer--;
+        if ((this->timer == 0) && (torchType != 0)) {
+            sLitTorchCount--;
+        }
+    }
+    if (this->timer != 0) {
+        if ((this->timer < 0) || (this->timer >= 20)) {
+            lightRadius = 200;
+        } else {
+            lightRadius = (this->timer * 200.0f) / 20.0f;
+        }
+        lightColorY = (u8)(Math_Rand_ZeroOne() * 127.0f) + 128;
```

Comment:
`lightColorY`? more like `lightColorRG`?

Reply:
This is the color luminosity, which I screwed up on the abbreviation (Y is luma, not luminosity). But looking up more color terms, just calling it `brightness` or `luminosity` is probably best.